File: frontend/lib/api/applicant-status.ts
import { api } from "@/lib/api-client";

import BackendServer from "@/constant/server-address";

const APPLICANT_STATUS_ENDPOINT = `${BackendServer}/api/ApplicantStatus`;

export interface ApplicantStatusItem {
  statusId: number;
  statusName: string;
  description: string;
  color: string;
  sortOrder: number;
  isActive: boolean;
  createdAt: string;
}

export const getApplicantStatuses = async (): Promise<
  ApplicantStatusItem[]
> => {
  return api.get<ApplicantStatusItem[]>(APPLICANT_STATUS_ENDPOINT);
};

File: backend/ai-recruitment/Features/ApplicationStatuses/Model/ApplicantStatus.cs
using ai_recruitment.Features.Candidates.model;
using System.ComponentModel.DataAnnotations;

namespace ai_recruitment.Features.ApplicationStatuses.Model
{
    public class ApplicantStatus
    {
        [Key]
        public int StatusId { get; set; }
        public required string StatusName { get; set; }
        public string? Description { get; set; }
        public string? Color { get; set; }
        public int SortOrder { get; set; }
        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        //public ICollection<Applicant> Applicants { get; set; } = new List<Applicant>();

    }
}
